x/axelarcork/types/msgs.go

```diff
@@ -0,0 +1,149 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	_ sdk.Msg = &MsgScheduleAxelarCorkRequest{}
+)
+
+const (
+	TypeMsgScheduleCorkRequest = "cork_schedule"
+	TypeMsgRelayCorkRequest    = "cork_relay"
+	TypeMsgBumpCorkGasRequest  = "cork_bump_gas"
+)
+
+////////////////////////////
+// MsgScheduleAxelarCorkRequest //
+////////////////////////////
+
+// NewMsgScheduleCorkRequest return a new MsgScheduleAxelarCorkRequest
+func NewMsgScheduleCorkRequest(body []byte, address common.Address, blockHeight uint64, signer sdk.AccAddress) (*MsgScheduleAxelarCorkRequest, error) {
+	return &MsgScheduleAxelarCorkRequest{
+		Cork: &Cork{
+			EncodedContractCall:   body,
+			TargetContractAddress: address.String(),
+		},
+		BlockHeight: blockHeight,
+		Signer:      signer.String(),
+	}, nil
+}
+
+// Route implements sdk.Msg
+func (m *MsgScheduleAxelarCorkRequest) Route() string { return ModuleName }
+
+// Type implements sdk.Msg
+func (m *MsgScheduleAxelarCorkRequest) Type() string { return TypeMsgScheduleCorkRequest }
+
+// ValidateBasic implements sdk.Msg
+func (m *MsgScheduleAxelarCorkRequest) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Signer); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, err.Error())
+	}
+
+	return m.Cork.ValidateBasic()
+}
+
+// GetSignBytes implements sdk.Msg
+func (m *MsgScheduleAxelarCorkRequest) GetSignBytes() []byte {
+	panic("amino support disabled")
```

Comment:
We should probably implement `GetSignBytes` for our messages so we have Ledger support.

Reply:
We don't support it in cork v1 or v2, but I followed the way gravity bridge does it, and also added the necessary codec.go modifications